File: app/src/main/java/com/example/projectappdoctruyen/Service/APIRetrofitClient.java
package com.example.projectappdoctruyen.Service;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class APIRetrofitClient {
    private static Retrofit retrofit = null;

    public static Retrofit getClient(String baseURL){
        if (retrofit == null){
            retrofit = new Retrofit.Builder()
                    .baseUrl(baseURL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }

}

File: app/src/main/java/com/example/projectappdoctruyen/Service/APIRetrofitClient.kt
package com.example.projectappdoctruyen.Service

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object APIRetrofitClient {
    private var retrofit: Retrofit? = null
    fun getClient(baseURL: String?): Retrofit? {
        if (retrofit == null) {
            retrofit = Retrofit.Builder()
                    .baseUrl(baseURL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
        }
        return retrofit
    }
}